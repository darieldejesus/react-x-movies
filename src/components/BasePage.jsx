import React from 'react';
import PropTypes from 'prop-types';
import Styled from 'styled-components';

import NavMenu from './NavMenu/';

const TopSection = Styled.section`
  background: #31353d;
  height: 200px;
`;

const ContentContainer = Styled.section`
  background: #fff;
  margin: 0 auto;
  max-width: 1024px;
  padding: 0 30px;
`;

class BasePage extends React.Component {
  render() {
    return [
      <TopSection key="topSection">
        {/* Logo should be here */}
        {/* Search input should be here */}
      </TopSection>,
      <ContentContainer key="contentSection">
        <NavMenu />
        {this.props.children}
      </ContentContainer>,
    ];
  }
}

BasePage.propTypes = {
  children: PropTypes.node.isRequired,
};

export default BasePage;
