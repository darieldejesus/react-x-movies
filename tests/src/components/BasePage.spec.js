import React from 'react';
import BasePage from '../../../src/components/BasePage';

describe('BasePage', function () {
  beforeEach(() => {
    this.wrapper = shallow(<BasePage><div>Yeey, I am a child</div></BasePage>);
  });

  test('should renders the BasePage component', () => {
    expect(this.wrapper).toMatchSnapshot();
  });
});
