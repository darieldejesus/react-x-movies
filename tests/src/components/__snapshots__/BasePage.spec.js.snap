// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasePage should renders the BasePage component 1`] = `
Array [
  <styled.section
    key="topSection"
  />,
  <styled.section
    key="contentSection"
  />,
]
`;
